packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/class_filter_dialog.dart

```diff
@@ -0,0 +1,159 @@
+
+  final ValueNotifier<ClassFilter> classFilter;
+
+  @override
+  State<ClassFilterDialog> createState() => _ClassFilterDialogState();
+}
+
+class _ClassFilterDialogState extends State<ClassFilterDialog> {
+  bool _initialized = false;
+  bool _showHelp = false;
+  late String _rootPackage;
+
+  late ClassFilterType _type;
+  final _except = TextEditingController();
+  final _only = TextEditingController();
+
+  @override
+  void initState() {
+    super.initState();
+    unawaited(_initialize());
+  }
+
+  Future<void> _initialize() async {
+    assert(!_initialized);
+    _rootPackage = await tryToDetectRootPackage() ?? '';
+    _loadStateFromFilter(widget.classFilter.value);
+    setState(() => _initialized = true);
+  }
+
+  @override
+  void didUpdateWidget(covariant ClassFilterDialog oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.classFilter.value != widget.classFilter.value) {
+      _loadStateFromFilter(widget.classFilter.value);
+    }
+  }
+
+  void _loadStateFromFilter(ClassFilter filter) {
+    _type = filter.filterType;
+    _except.text = filter.except;
+    _only.text = filter.only ?? _rootPackage;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    final textFieldLeftPadding = scaleByFontFactor(40.0);
+    final itemSpacing = scaleByFontFactor(28.0);
+
+    void onTypeChanged(ClassFilterType? type) => setState(() => _type = type!);
+
+    return DevToolsDialog(
+      title: dialogTitleText(theme, 'Filter Classes and Packages'),
+      content: !_initialized
+          ? Progress(size: 30.0)
+          : Container(
+              width: defaultDialogWidth,
+              child: Column(
+                mainAxisSize: MainAxisSize.min,
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  IconLabelButton(
+                    tooltip: _showHelp ? 'Hide help' : 'Show help',
+                    icon: _showHelp
+                        ? Icons.keyboard_arrow_up
+                        : Icons.keyboard_arrow_down,
+                    label: 'Help',
+                    onPressed: () => setState(() => _showHelp = !_showHelp),
+                  ),
+                  if (_showHelp) ...[
+                    const SizedBox(height: denseSpacing),
+                    const Text('Choose and customize the filter.\n'
+                        'List full or partial class names separated by new lines. For example:\n\n'
+                        '  package:myPackage/src/myFolder/myLibrary.dart/MyClass\n'
+                        '  MyClass\n'
+                        '  package:myPackage/src/\n\n'
+                        'Specify:\n'
+                        '  - ${ClassFilter.coreLibrariesAlias} for core classes without package prefix\n'
+                        '  - ${ClassFilter.dartAndFlutterLibrariesAlias} for "dart." and "package:" libraries published by Dart and Flutter orgs.'),
+                  ],
```

Comment:
additionally, pull all this help info into a private widget inside this file. This build method is getting a bit long